Source/DataSources/GeometryVisualizer.js

```diff
@@ -376,8 +447,17 @@ define([
 
         if (updater.fillEnabled) {
             if (updater.onTerrain) {
+                var zIndex = updater.zIndex;
+                var batchArray = this._groundColorBatches[zIndex];
+                if (!defined(batchArray)) {
+                    batchArray = createGroundColorBatch(this._orderedGroundPrimitives.add(new PrimitiveCollection()));
```

Comment:
Right now I create a three `new StaticGroundGeometryColorBatch` (one for each classification type) and one `new PrimitiveCollection()` for each zIndex.  
And I never remove them (the PrimitiveCollections get removed and destroyed on `GeometryVisualizer.destroy`)

Do you think this is a problem?  I didn't think it would be because the `zIndex` has to be `ConstantProperty` so we shouldn't end up with empty primitive collections unless someone changes the `zIndex` value.

Instead of creating a `StaticGroundGeometryColorBatch` for each classification type, do you think I should create them only if that classification type gets used?

Reply:
Just because it's a constant property doesn't mean it can't change, it just means it won't change with respect to simulation time.  I think we should try and delay creating anything until we actually need it, perhaps the key to `_groundColorBatches` can be the zIndex and type together?

Is there a reason we also can't remove it if/when it's empty?

Reply:
Another thing about this feature that I don't think was made clear is that zIndex only works for completely non-time-varying ground primitives.  It has nothing to do with zIndex being constant or not.  The way you have the code written here, if color/positions/etc.. is time-dynamic, z-indexing doesn't work. (Unless I'm missing something?)

Reply:
> I think we should try and delay creating anything until we actually need it, perhaps the key to _groundColorBatches can be the zIndex and type together?

Good idea, I think that approach will work

> Is there a reason we also can't remove it if/when it's empty?

I'm not sure exactly how to figure out whether a batch is empty. `StaticGroundGeometryColorBatch` doesn't have a length property or anything.

> Another thing about this feature that I don't think was made clear is that zIndex only works for completely non-time-varying ground primitives.

Yes that's true.  I have no idea how to maintain zIndex across the `StaticGroundGeometryColorBatch` and the `DynamicGeometryBatch` without making bigger changes.  There would have to be some class that aggregates both and manages the primitives.

Reply:
> I have no idea how to maintain zIndex across the StaticGroundGeometryColorBatch and the DynamicGeometryBatch without making bigger changes.

Actually, this is probably something we'll need anyway for when the ground primitive material changes come in.  I'll have to think about this. I have an idea for what I want to do.

Reply:
OK, it might be worth holding off this PR until after ground primitive materials are ready then.  I think this initial attempt was definitely worth it; but to avoid churn it might be better to do everything at once.